unoR3/unoR3.ino

```diff
@@ -1,39 +1,47 @@
+#include <HomeAutomationLight.h>
+#include <HomeAutomationTimer.h>
 #include <dht.h>
 
 dht DHT;
 
-#define DHT11_PIN 5
-#define HEATER 7
-#define WINDOW_SENSOR_PIN 8
-#define MAINLIGHT 11
-#define MIRRORLIGHT 10
-#define LAMPLEFT 2
-#define LAMPRIGHT 3
-#define DOOR_SENSOR_PIN 9
+const int dht_pin = 5;
+const int heater_pin = 7;
+const int window_sensor_pin = 8;
+const int main_light_pin = 11;
+const int mirror_lamp_pin = 10;
+const int left_lamp_pin = 2;
+const int right_lamp_pin = 3;
+const int door_sensor_pin = 9;
 
 // overrides
 const int OVERRIDE_ON = 0;
 const int OVERRIDE_OFF = 1;
 const int DONT_OVERRIDE = 2;
+// lights
+HomeAutomationLight main_light(main_light_pin, false); // active low
+HomeAutomationLight mirror_light(mirror_lamp_pin, false);  // active low
+HomeAutomationLight left_lamp(left_lamp_pin);
+HomeAutomationLight right_lamp(right_lamp_pin);
+
+boolean door_light_active = true; // controls if the mirror light should come on when the door opens
+unsigned long mirror_on_time_ms = 5000;
+// create a timer to manage the turning off of the mirror light after 5 seconds
+auto mirror_timer = HomeAutomationTimer(mirror_on_time_ms, 
+                                        [&mirror_light](){ mirror_light.on(); },   // call mirror_light.on when triggered
+                                        [&mirror_light](){ mirror_light.off(); });  // call mirror_light.off when time expires 
```

Comment:
don't worry too much about understanding these 3 lines. I don't fully understand them, I got some help from a friend in writing them.

Basically we create a timer object and tell it the time it should run for (`mirror_on_time_ms`) and give it two actions, one to call when the timer gets activated (`mirror_light.on()`) and one to call with the time runs out (`mirror_light.off()`) the rest is just faff needed to get it to work